gtsfm/loader/one_d_sfm_loader.py

```diff
@@ -0,0 +1,163 @@
+    Tuytelaars, T. (eds) Computer Vision – ECCV 2014. ECCV 2014. Lecture Notes in Computer Science, vol 8691. Springer,
+    Cham. https://doi.org/10.1007/978-3-319-10578-9_5
+    [2] J. L. Schönberger and J. -M. Frahm, "Structure-from-Motion Revisited," 2016 IEEE Conference on Computer Vision
+    and Pattern Recognition (CVPR), 2016, pp. 4104-4113, doi: 10.1109/CVPR.2016.445.
+    """
+
+    def __init__(
+        self,
+        folder: str,
+        image_extension: str = "jpg",
+        max_resolution: int = 640,
+        max_num_imgs: int = 0,
+    ) -> None:
+        """Initializes to load from a specified folder on disk.
+
+        Args:
+            folder: the base folder which contains image sequence.
+            image_extension: file extension for the image files. Defaults to 'jpg'.
+            max_resolution: integer representing maximum length of image's short side, i.e.
+                the smaller of the height/width of the image. e.g. for 1080p (1920 x 1080),
+                max_resolution would be 1080. If the image resolution max(height, width) is
+                greater than the max_resolution, it will be downsampled to match the max_resolution.
+            max_num_imgs: integer representing max number of images to process in the sequence,
+                0 or negative values means to process all the images.
+        """
+        super().__init__(max_resolution=max_resolution)
+
+        # Fetch all the file names in /images folder.
+        search_path = os.path.join(folder, "images", f"*.{image_extension}")
+
+        # NOTE(yanwei.du) Currently we only process images with valid EXIF data.
+        (
+            self._image_paths,
+            self._num_exif_imgs,
+            self._num_all_imgs,
+        ) = self.get_images_with_valid_exif(search_path, max_num_imgs)
+        self._num_imgs = len(self._image_paths)
+
+        print(f"Selected {self._num_imgs} out of {self._num_exif_imgs} images with valid exif.")
+
+        if self._num_imgs == 0:
+            raise RuntimeError(f"Loader could not find any images with the specified file extension in {search_path}")
+
+    def get_images_with_valid_exif(self, search_path: str, max_num_imgs: int) -> Tuple[List[str], int, int]:
+        """Return a subset of images with valid exif.
+
+        Args:
+            search_path: image sequence search path.
+            max_num_imgs: the maximum number of images to process.
+
+        Returns:
+            Tuple[
+                List of selected image path.
+                The number of images with valid exif data.
+                The number of all the images.
+            ]
+        """
+        all_image_paths = glob.glob(search_path)
+        num_all_imgs = len(all_image_paths)
+        exif_image_paths = []
+        for single_img_path in all_image_paths:
+            # Drop images without valid exif data.
+            if io_utils.load_image(single_img_path).get_intrinsics_from_exif():
```

Comment:
might be a littler clearer to show we're checking for None, and not for a boolean, as

```python
if io_utils.load_image(single_img_path).get_intrinsics_from_exif() is not None:
    exif_image_paths.append(single_img_path)
```

Reply:
Done